### PR TITLE
Remove `JsonbPostgreSQL95Dialect` from quarkus.

### LINKED ISSUE #16496: Remove `JsonbPostgreSQL95Dialect` from `DatabasePropertyMappers`
### Description

With hibernate 6 there is new way to register custom functions, therefore there is no need to specify custom dialect (``JsonbPostgreSQL95Dialect``). 

There is: 
https://github.com/keycloak/keycloak/blob/dceb2f96b274b83c764c2ba8fd98dab16980f757/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java#L189

It should be removed from `DatabasePropertyMappers` and replaced by detecting correct dialect (`PostgreSQLDialect` or `CockroachDialect`). 

Determining correct dialect is important as there is a need to register slightly different function for postgres and cocroach and dialect'd be used for detection.

Note: It'd probably require adding new config options when starting quarkus with map storage.

### PR DESCRIPTION
**Pull request by vramik**
EDIT: 

Fixes #16496

### CODE REVIEW

File: quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java [hunk 1]
```diff
@@ -324,6 +326,15 @@ public String getProvider() {
             .hidden()
             .build();
 
+    public static final Option<String> STORAGE_JPA_DB = new OptionBuilder<>("storage-jpa-db", String.class)
+            .category(OptionCategory.STORAGE)
+            .defaultValue(Database.Vendor.POSTGRES.toString())
+            .expectedValues(Database::getAvailableMapStoreAliases)
+            .description("The database vendor for jpa map storage.")
+//            .hidden() // todo ??
```
Comment (pedroigor) [inline quarkus/config-api/src/main/java/org/keycloak/config/StorageOptions.java:334]: The decision to mark it as a `hidden` option depends on how you expect users to switch databases. If you are not sure about making it fully supported, marking it as `hidden` makes sense.

**Comment by vramik** [comment]
The PR is currently blocked by unavailability of Hiberante 6 in quarkus3. Jpa map store in quarkus3 branch is not compatible with Hibernate 5 ( due to changes done by https://github.com/keycloak/keycloak/issues/16335 ), therefore it's not possible to start quarkus with changes from this PR.

@ahus1 @hmlnarik Could you please take a look whether it goes the right direction and the code makes sense? Thank you.

**Review by ahus1 — COMMENTED** [review]
LGTM with a minor comment to remove that commented out line, as this option should not be hidden. Once there is a change in the CLI, all of the `HelpCommandDistTest.testStartHelpAll.unix.approved.txt` et al need to be updated to make those tests pass.

I see that this change doesn't break the legacy store, this starts just fine. 

So the only reason to not merge it would be that we haven't tested it. As this is only about merging it to the Q3 branch, I'd be ok with merging it now and fixing any problems with H6 later once it appears in the respective Q3 alpha (which we'll have some probably anyway). This would avoid rebasing this PR again and again.

**Comment by vramik** [comment]
Thank you @ahus1!

> LGTM with a minor comment to remove that commented out line, as this option should not be hidden. 

+1 removed

> Once there is a change in the CLI, all of the `HelpCommandDistTest.testStartHelpAll.unix.approved.txt` et al need to be updated to make those tests pass.

Thank you for pointing that out, I wasn't aware of it. Updated

> I see that this change doesn't break the legacy store, this starts just fine.
> 
> So the only reason to not merge it would be that we haven't tested it. As this is only about merging it to the Q3 branch, I'd be ok with merging it now and fixing any problems with H6 later once it appears in the respective Q3 alpha (which we'll have some probably anyway). This would avoid rebasing this PR again and again.

That would work for me as well. If we agree on it I'll switch the PR to `Ready for review`

**Comment by vramik** [comment]
Once https://github.com/keycloak/keycloak/pull/17327 is available in `quarkus3` branch this PR should:

1. update `DatabasePropertyMappers`

```java
private static Optional<String> transformDialect(Optional<String> db, ConfigSourceInterceptorContext context) {
        if (isJpaStore()) {
            return Database.getDialect(getJpaStoreDbVendor().name().toLowerCase());
        }
...
```

2. update dialect of `Database.Vendor.COCKROACH` enum to `org.hibernate.dialect.CockroachDialect`

**Review by ahus1 — APPROVED** [review]
Thank you for preparing this PR - this is ready to be merged. Thanks!

**Review by mhajas — APPROVED** [review]
Approving per @ahus1 's approval

**Pull request merged**